bin/compile

```diff
@@ -123,24 +126,17 @@ if [ -x .flutter/bin/flutter ]; then
   cp -R $SOURCE_DIR/.pub-cache $CACHE_DIR/.pub-cache
 fi
 
+# Move built website inside deploy dir before cleaning files
+mv $SOURCE_DIR/build/web/* $DEPLOY_DIR
+
 # Check  FLUTTER_CLEANUP var to delete all files or keep it.
 if [ "$FLUTTER_CLEANUP" != false ]; then
   print "Cleaning up Flutter source files and the SDK"
 
-  mkdir -p $SOURCE_DIR/TO_DELETE
-
-  mv !("TO_DELETE") TO_DELETE
-  mv TO_DELETE/build/web/* $DEPLOY_DIR
-
-  rm -rf TO_DELETE
-fi
+  rm -rf $SOURCE_DIR 
 
-# heroku-buildpac-static compatibility
-if [ ! -f $BUILD_DIR/static.json ]; then
-  echo "{}" > $BUILD_DIR/static.json
-fi
+else
 
-if [ "$FLUTTER_CLEANUP" == false ]; then
```

Comment:
The code below is supposed to only be executed when `FLUTTER_CLEANUP` is `false`, because if it's not, then Dart/Flutter SDKs are supposed to be removed, so setting them up wouldn't make sense (since they would be gone).

By removing `if [ "$FLUTTER_CLEANUP" == false ]; then` and merging the code block with an earlier `if [ "$FLUTTER_CLEANUP" != false ]; then` block (notice that one has `==` and the other has `!=`) you cause the behavior to be the opposite of the intended behavior - Flutter & Dart are only set up in cases when they were supposed to be removed but are NOT set up in cases when  they are supposed to stay. This in turn breaks the CI support (which [sets `FLUTTER_CLEANUP` to `false` internally](https://github.com/EE/heroku-buildpack-flutter-light/blob/master/bin/test-compile)).

Reply:
I don't see right now how the refactor doesn't follow the original logic, but I don't code daily in bash and I have already many times being surprised by its quirks so I'll check this out at some later time :) Thanks!